### PR TITLE
chore(mojaloop/#3451): nodejs upgrade

### PR DESCRIPTION
**Pull request by oderayi**
chore(mojaloop/#3451): NodeJS version upgrade - https://github.com/mojaloop/project/issues/3451
- Upgraded NodeJS from v16 to v18 LTS
- Updated CI config to use NodeJS latest LTS version
- Updated CI config to display runtime versions for each job
- Upgraded NPM dependencies
- Fixed CI bugs caused by the NodeJS upgrade
- Upgraded CircleCI orbs versions
- Replaced `audit-resolver` with `audit-ci`
-  Updated CI pipeline use NVM to select node version from` .nvmrc` file

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -7,9 +7,9 @@ version: 2.1
...
 ##
 orbs:
-  slack: circleci/slack@4.9.3 # Ref: https://github.com/mojaloop/ci-config/tree/master/slack-templates
+  slack: circleci/slack@4.12.5 # Ref: https://github.com/mojaloop/ci-config/tree/master/slack-templates
   pr-tools: mojaloop/pr-tools@0.1.10 # Ref: https://github.com/mojaloop/ci-config/
-  gh: circleci/github-cli@2.1.0
+  gh: circleci/github-cli@2.2.0
 
 ##
 # defaults
```
Comment (kleyow) [inline .circleci/config.yml:15]: You're using the sdk-scheme-adapter nodejs upgrade as a template right?

Can we keep variables in sync?

ex: defaults_docker_Dependencies: &defaults_docker_Dependencies
https://github.com/mojaloop/sdk-scheme-adapter/pull/453/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R23

Here and throughout.

File: .circleci/config.yml [hunk 2]
```diff
@@ -59,6 +58,58 @@ defaults_configure_git: &defaults_configure_git
     git config user.email ${GIT_CI_EMAIL}
     git config user.name ${GIT_CI_USER}
 
+defaults_configure_nvm: &defaults_configure_nvm
```
Comment (kleyow) [inline .circleci/config.yml:61]: I see you merged defaults_configure_nvm: &defaults_configure_nvm and defaults_configure_nvmrc: &defaults_configure_nvmrc from the sdk-scheme-adapter together. 

I'm fine with either grouping but just a reminder that we'll need to keep the structure in sync.

File: .circleci/config.yml [hunk 3]
```diff
@@ -138,7 +210,7 @@ jobs:
           command: npm -s run test:xunit
       - store_artifacts:
           path: ./test/results
-          prefix: test
+          destination: test
```
Comment (kleyow) [inline .circleci/config.yml:213]: Is it that prefix is deprecated and destination is the new key?

Comment (oderayi) [reply .circleci/config.yml:213]: Correct @kleyow.

File: .circleci/config.yml (hunk 2 above)
Comment (oderayi) [reply .circleci/config.yml:61]: @kleyow Here we're only interested in `defaults_configure_nvm: &defaults_configure_nvm`.

Comment (kleyow) [reply .circleci/config.yml:61]: Ah my mistake.

**Review by kleyow — APPROVED** [review]

**Pull request merged**